engine/src/cmd/collide_map.cpp

```diff
@@ -36,27 +39,23 @@ void CollideArray::erase(iterator target) {
     count -= 1;
     if (target >= this->begin() && target < this->end()) {
         target->radius = 0;
-        target->ref.unit = NULL;
+        target->ref.unit = nullptr;
         size_t diff = (target - this->begin());
         if (this->unsorted.size() > diff) {
             //for secondary collide arrays that have no unsorted array
             iterator tmp = &*(this->unsorted.begin() + diff);
             tmp->radius = 0;
-            tmp->ref.unit = NULL;
+            tmp->ref.unit = nullptr;
         }
         return;
-    } else if (target == NULL) {
+    } else if (target == nullptr) {
         return;
     } else {
-        CollidableBackref *targ = static_cast< CollidableBackref * > (&*target);
-        std::list<CollidableBackref> *targlist = &toflattenhints[targ->toflattenhints_offset];
-        std::list<CollidableBackref>::iterator endlist = targlist->end();
-        for (std::list<CollidableBackref>::iterator i = targlist->begin(); i != endlist; ++i) {
-            if (&*i == target) {
-                targlist->erase(i);
-                return;
-            }
-        }
+        CollidableBackref *targ = vega_dynamic_cast_ptr<CollidableBackref>(&*target);
+        std::list<CollidableBackref> *targlist = &toflattenhints.at(targ->toflattenhints_offset);
+        auto first_to_remove = std::stable_partition(targlist->begin(), targlist->end(), [target](CollidableBackref &backref) { return &*backref != target; });
+//        std::for_each(first_to_remove, targlist->end(), [](CollidableBackref &backref) { delete backref; });
+        targlist->erase(first_to_remove, targlist->end());
```

Comment:
Please note that this change may affect game behavior, unlike most of the changes in this PR. However, I think this block of changes is almost mandatory for correctness.

Reply:
yeah, the code on the left certainly isn't right. I did some similar change in @royfalk 's SDL2 PR....https://github.com/royfalk/Vega-Strike-Engine-Source/pull/1/files#r1292449185